src/services/homework/hooks/submissions.js

```diff
@@ -130,15 +130,21 @@ const insertSubmissionData = (hook) => {
 const insertHomeworkData = (hook) => {
 	const homeworkId = hook.data.homeworkId || (hook.data.submission || {}).homeworkId;
 	if (homeworkId) {
+		const userId = hook.params.account.userId.toString();
 		const homeworkService = hook.app.service('/homework');
-		return homeworkService.get(homeworkId, { account: { userId: hook.params.account.userId } })
+		return homeworkService.get(homeworkId, {
+			account: { userId },
+			query: {
+				$populate: ['courseId'],
+			},
+		})
 			.then((homework) => {
 				hook.data.homework = homework;
 				// isTeacher?
 				hook.data.isTeacher = false;
 				if ((hook.data.homework.teacherId == hook.params.account.userId)
-						|| (hook.data.homework.courseId.teacherIds || []).includes(hook.params.account.userId)
-						|| (hook.data.homework.courseId.substitutionIds || []).includes(hook.params.account.userId)) {
+						|| (hook.data.homework.courseId.teacherIds || []).some((el) => el.toString() === userId) //
```

Comment:
same like above... i like userId  here, but how about currentUserId and courseTeacherId instead of el?